app/models/base_interactive.rb

```diff
@@ -19,9 +19,30 @@ def portal_hash
     iframe_data
   end
 
+  def parsed_authored_state
+    JSON.parse(authored_state).symbolize_keys rescue {}
+  end
+
   def report_service_hash
-    {
-      type: 'iframe_interactive',
+    # Expected metadata format can be checked in lara-typescript/interactive-api-client/metadata-types.ts:
+    # IAuthoring<...>Metadata interfaces.
+    # Metadata is simply provided as a part of authored state. It's optional and everything should work if there's
+    # no metadata defined or authored state is empty.
+    metadata = parsed_authored_state
+
+    type = metadata[:questionType] || "iframe_interactive"
+
+    result = {
+      # type can be overwritten by metadata[:questionType] prop (e.g. to "open_response").
+      # Otherwise, the default "iframe_interactive" will be used.
+      type: type,
+      # These properties are defined in IAuthoringMetadataBase:
+      subtype: metadata[:subtype],
+      required: metadata[:required],
+      prompt: metadata[:prompt],
+      # These properties are stored in LARA. They're basic interactive properties. Some of them might be unused by
+      # Report or Portal when interactive pretends to be a basic question type. But these services might be extended
+      # to show both basic question answer and optionally provide iframe report view.
```

Comment:
I've added it in the last commit. 

We just never used this authored state in question level I think. Now, we still don't use it, as we copy some properties to top-level data (like prompt, required etc.). Once I've added the authored state as you suggested, we could just make assumptions about authored state in the report code instead and read the prompt directly from there. It can make sense once we deal only interactives (no more old-style questions anymore). 

Then, LARA wouldn't have to care about data format and the report code could import all the interfaces from lara-interactive-api NPM package.

Reply:
True except for the LARA report, but hopefully we can replace that with the portal-report.

Reply:
Right. Although I guess the LARA report only uses `report_state` to initialize iframes (at least in its current form), so I don't think it cares about any metadata / authored state format.